src/modal/modal-stack.ts

```diff
@@ -150,4 +163,17 @@ export class NgbModalStack {
     this._modalRefs.push(ngbModalRef);
     ngbModalRef.result.then(unregisterModalRef, unregisterModalRef);
   }
+
+  private _registerWindowCmpt(ngbWindowCmpt: ComponentRef<NgbModalWindow>) {
+    this._windowCmpts.push(ngbWindowCmpt);
+    this._activeWindowCmptHasChanged.next();
+
+    ngbWindowCmpt.onDestroy(() => {
+      const index = this._windowCmpts.indexOf(ngbWindowCmpt);
+      if (index > -1) {
```

Comment:
I copied this check from `_registerModalRef`. Is it really necessary to check the existence of an element before removing it from an array?

Reply:
With `splice()`yes it is needed, because `splice( negative-index-here, ...)` will remove from the end, e.g. : `[1, 2 , 3].splice(-1, 1)` -> `[1,  2]`

As a sidenote, I think we could only fire `this._activeWindowCmptHasChanged.next();` if we remove a componentRef. If nothing is removed it is not needed.

Reply:
> As a sidenote, I think we could only fire this._activeWindowCmptHasChanged.next(); if we remove a componentRef. If nothing is removed it is not needed.

I subscribe to `_activeWindowCmptHasChanged` in constructor in order to call `ngbFocusTrap` on the top/activate/latest (you get the idea) `WindowCmpt`. So it should fire on `push`.

Reply:
I was just talking about moving the call inside the `if `:

```typescript
ngbWindowCmpt.onDestroy(() => {
    const index = this._windowCmpts.indexOf(ngbWindowCmpt);
    if (index > -1) {
        this._windowCmpts.splice(index, 1);
        this._activeWindowCmptHasChanged.next();
    }
    // this._activeWindowCmptHasChanged.next();  move it inside the upper if
});

Reply:
oh, sure, I misunderstood you.